### DISCUSSION TITLE
Using firejail on a terminal emulator that allows "sudo"

### DISCUSSION THREAD

**Discussion by spatial25** [netblue30/firejail#5483] [Q&A] [answered]
I'm trying to use firejail on my terminal emulator and it works but I am not sure how to allow it to use sudo. Right now I get 'bash: permission denied' I don't see a profile for the terminal application and am thinking that I can make a copy of the default firejail profile and make some changes to it but I am unsure on which ones to make. Any help out there?

**Comment by ghost** [comment, marked as answer]
Sandboxing terminal emulators is notoriously complex. That's why we tend to not have profiles for them, nor are any enabled by default in firecfg for example. Emulators which run as servers can escape sandboxes: see [disable-common.inc](https://github.com/netblue30/firejail/blob/master/etc/inc/disable-common.inc#L558-L575).

On top of the above, allowing `sudo` is a rather tedious work of trial-and-error and usually ends up with a very weak sandbox. Note that this is not to discourage you from trying, just adding some context so you're aware of potential pitfalls. Here's a list of some of firejail options to attend to:

- caps
- nonewprivs
- protocol
- seccomp
- private-bin
- private-lib

IMO it's not worth the trouble. Try to enable and tighten the firejail profiles for as much of the applications/software you use instead. Just my two cents.

**Comment by rusty-snake** [comment]
Unless you have special usecases (I can not imagination of any than requires sudo), your terminal-emulator is a trusted domain.
